client/src/components/ActivityDetails/index.js

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from "react";
+import axios from 'axios';
+import {StyledLiTitle, StyledDetailLi, StyledUl} from "./index.style";
+
+class ActivityDetails extends Component {
+
+  constructor(props){
+    super(props);
+    this.state = {
+      renderedDetails: <div>Loading...</div>,
+    };
+    this.getActivityDetailsById = this.getActivityDetailsById.bind(this);
+    this.getTrainingDetails = this.getTrainingDetails.bind(this);
+  }
+
+  componentDidMount(){
+    this.getActivityDetailsById();
+  }
+
+
+  getActivityDetailsById(){
+    const { activityId } = this.props;
+      axios.get('/activities/'+activityId)
+        .then(({ data }) => {
+          if(data.success){
+            this.setState({ renderedDetails:  this.getTrainingDetails(data.data)});
+          } else {
+            alert("Error: "+data.err);
+          }
+        }).catch((err) => {
+          alert("Error: " + err.message)
+      });
+  }
+
+  dateParser(date){
+    //console.log(new Date(date).toLocaleString());
+    const isoDate = ((new Date(date).toLocaleString()).split(',')[0]).split('/');
+    return (isoDate[2] + '/' + isoDate[0] + '/' + isoDate[1]);
+  }
+
+
+getTrainingDetails(detailsList) {
+  const renderedDetails = Object.keys(detailsList).map((k) => {
```

Comment:
What does k mean here?

Reply:
Means key of the object, will change it to `key`.